File: src/Application Specific/Classes/Entity Collections/Jobs/EngineerVisit/EngineerVisitNCCGSR/EngineerVisitNCCGSRValidator.vb
Namespace Entity

Namespace EngineerVisitNCCGSRs

Public Class EngineerVisitNCCGSRValidator
    'make sure that contact object is valid
    Inherits BaseValidator

    Public Sub Validate(ByVal oEngineerVisitNCCGSR As EngineerVisitNCCGSR)

        'make sure that contact object is valid
        If oEngineerVisitNCCGSR.Errors.Count > 0 Then
            Dim de As DictionaryEntry
            For Each de In oEngineerVisitNCCGSR.Errors
                Me.AddCriticalMessage(CStr(de.Value))
            Next
        End If

        
	
        If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
            Throw New ValidationException(Me)
        End If

    End Sub

End Class

End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Jobs/EngineerVisit/EngineerVisitNCCGSR/EngineerVisitNCCGSRValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace EngineerVisitNCCGSRs
    {
        // make sure that contact object is valid
        public class EngineerVisitNCCGSRValidator : BaseValidator
        {
            public void Validate(EngineerVisitNCCGSR oEngineerVisitNCCGSR)
            {

                // make sure that contact object is valid
                if (oEngineerVisitNCCGSR.Errors.Count > 0)
                {
                    DictionaryEntry de;
                    foreach (var de in oEngineerVisitNCCGSR.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}